extension/lib/bigcommerce/RequestRepository.js

```diff
@@ -0,0 +1,92 @@
+const BigCommerceLogger = require('./Logger')
+
+class BigCommerceRequestRepository {
+  /**
+   * @param {BigCommerce} client Api V3 client
+   * @param {Logger} logger context.log
+   */
+  constructor (client, logger) {
+    this.client = client
+    this.logger = logger
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  get (path) {
+    return this.request('get', path)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  post (path, data, obfuscatedData = null) {
+    return this.request('post', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  put (path, data, obfuscatedData = null) {
+    return this.request('put', path, data, obfuscatedData)
+  }
+
+  /**
+   * @param {string} path
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  del (path) {
+    return this.request('delete', path)
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} path
+   * @param {Object} [data]
+   * @param {Object} [obfuscatedData]
+   * @return {BigCommerceRedirectUrlsResponse}
+   */
+  async request (type, path, data = null, obfuscatedData = null) {
+    const requestData = this.getRequestData(type, path, data, obfuscatedData)
+    const logRequest = new BigCommerceLogger(this.logger)
+    const start = new Date()
+
+    try {
+      const response = await this.client.request(type, path, data)
+      logRequest.log(requestData, response, new Date() - start, 1)
+
+      return response
+    } catch (e) {
+      logRequest.log(requestData, e.toString(), new Date() - start, 0)
+
+      throw e
+    }
+  }
+
+  /**
+   * @param {string} type
+   * @param {string} path
+   * @param {Object} data
+   * @param {Object} obfuscatedData
+   * @return {Object}
+   */
+  getRequestData (type, path, data, obfuscatedData) {
+    if (obfuscatedData !== null) {
+      return { type, path, data: obfuscatedData }
```

Comment:
something went horribly wrong here. You just said screw the whole `data`, just give what's in the obfuscated object. E.g. if u have `{ username: test, password: kill123}` and obfuscated `{password: xxxxx}`, then you are ignoring username and any other variable.
I would recommend obfuscated to be an array instead though. It would be easier to just pass an array of properties that need obfuscating. I don't believe we care how they should be obfuscated, e.g. `['password']` instead of `{ password: 'XXXXX' }` 

So you can just do this afterwards:
```
obfuscated.forEach(field => { 
  if (data.hasOwnProperty(field)) {
    data[field] = 'XXXXX'
  }
})
```
p.s. test it plz

Reply:
Thought about this too.
But there is currently no use case for this (obfuscating the data partially) and I don't see one coming up.
So this looks like overengineering to me.